packages/fxa-auth-server/lib/payments/stripe-firestore.ts

```diff
@@ -211,6 +232,61 @@ export class StripeFirestore {
       .set(invoice, { merge: true });
   }
 
+  /**
+   * Insert an invoice record into Firestore under the customer's stripe id.
+   */
+  async insertPaymentMethodRecord(
+    paymentMethod: Partial<Stripe.PaymentMethod>
+  ) {
+    const customerSnap = await this.customerCollectionDbRef
+      .where('id', '==', paymentMethod.customer as string)
+      .get();
+    if (customerSnap.empty) {
+      throw newFirestoreStripeError(
+        `Customer ${paymentMethod.customer} was not found`,
+        FirestoreStripeError.FIRESTORE_CUSTOMER_NOT_FOUND
+      );
+    }
+
+    return customerSnap.docs[0].ref
+      .collection(this.paymentMethodCollection)
+      .doc(paymentMethod.id!)
+      .set(paymentMethod, { merge: true });
+  }
+
+  /**
+   * Insert a payment method record into Firestore under the customer's stripe id.
+   * If the customer does not exist, this will backfill the customer with all their
+   * subscriptions.
+   */
+  async insertPaymentMethodRecordWithBackfill(
+    paymentMethod: Partial<Stripe.PaymentMethod>
+  ) {
+    try {
+      await this.insertPaymentMethodRecord(paymentMethod);
+    } catch (err) {
+      if (err.name === FirestoreStripeError.FIRESTORE_CUSTOMER_NOT_FOUND) {
+        await this.fetchAndInsertCustomer(paymentMethod.customer as string);
```

Comment:
We'd still need to insert the payment method here correct?

Reply:
Yep, we sure do! Doh!